src/wp-includes/class-wp-scripts.php

```diff
@@ -714,6 +781,262 @@ public function in_default_dir( $src ) {
+	 *      @type string    $strategy     Optional. Values blocking|defer|async. Default 'blocking'.
+	 * }
+	 * @return array        Normalized $args array.
+	 */
+	private function get_normalized_script_args( $handle, $args = array() ) {
+		$default_args = array(
+			'in_footer' => false,
+			'strategy'  => 'blocking',
+		);
+
+		// Handle backward compatibility for $in_footer.
+		if ( true === $args ) {
+			$args = array( 'in_footer' => true );
+		}
+
+		return wp_parse_args( $args, $default_args );
+	}
+
+	/**
+	 * Get all dependents of a script.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $handle The script handle.
+	 * @return array Array of script handles.
+	 */
+	private function get_dependents( $handle ) {
+		// Check if dependents map for the handle in question is present. If so, use it.
+		if ( array_key_exists( $handle, $this->dependents_map ) ) {
+			return $this->dependents_map[ $handle ];
+		}
+
+		$dependents = array();
+
+		// Iterate over all registered scripts, finding dependents of the script passed to this method.
+		foreach ( $this->registered as $registered_handle => $args ) {
+			if ( in_array( $handle, $args->deps, true ) ) {
+				$dependents[] = $registered_handle;
+			}
+		}
+
+		// Add the handles dependents to the map to ease future lookups.
+		$this->dependents_map[ $handle ] = $dependents;
+
+		return $dependents;
+	}
+
+	/**
+	 * Determines if a script loading strategy is valid.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $strategy A script loading strategy.
+	 * @return bool True if the strategy is of an allowed type, false otherwise.
+	 */
+	private function is_valid_strategy( $strategy ) {
+		$allowed_strategies = array( 'blocking', 'defer', 'async' );
+
+		return in_array( $strategy, $allowed_strategies, true );
+	}
+
+	/**
+	 * Get the strategy assigned during script registration.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $handle The script handle.
+	 * @return string Strategy set during script registration. Empty string if none was set.
+	 */
+	private function get_intended_strategy( $handle ) {
+		$script_args = $this->get_data( $handle, 'script_args' );
+		$strategy    = isset( $script_args['strategy'] ) ? $script_args['strategy'] : '';
+
+		if ( $strategy && ! $this->is_valid_strategy( $strategy ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				sprintf(
+					/* translators: 1: $strategy, 2: $handle */
+					__( 'Invalid strategy `%1$s` defined for `%2$s` during script registration.' ),
+					$strategy,
+					$handle
+				),
+				'6.3.0'
+			);
+
+			return '';
+		}
+
+		return $strategy;
+	}
+
+	/**
+	 * Check if the strategy passed is a non-blocking strategy.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $strategy The strategy to check.
+	 * @return bool True if $strategy is one of the delayed strategy.
+	 */
+	private function is_non_blocking_strategy( $strategy ) {
+		$delayed_strategies = array( 'defer', 'async' );
+
+		return in_array( $strategy, $delayed_strategies, true );
```

Comment:
Maybe a tiny bit better.

```suggestion
		return in_array( 
			$strategy,
			array( 'defer', 'async' ),
			true
		);
```

Reply:
https://github.com/WordPress/wordpress-develop/pull/4391/files#r1180449601